ScopedStorage/app/src/main/java/com/samples/storage/mediastore/AddMediaViewModel.kt

```diff
@@ -0,0 +1,107 @@
+package com.samples.storage.mediastore
+
+import android.Manifest.permission.WRITE_EXTERNAL_STORAGE
+import android.app.Application
+import android.content.ContentValues
+import android.content.Context
+import android.content.pm.PackageManager
+import android.net.Uri
+import android.os.Build
+import android.provider.MediaStore
+import androidx.core.content.ContextCompat.checkSelfPermission
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+
+class AddMediaViewModel(application: Application) : AndroidViewModel(application) {
+
+    private val context: Context
+        get() = getApplication()
+
+    val isPermissionGranted: Boolean
+        get() = canWriteInMediaStore(context)
+
+    // We keep the current Media in the viewmodel to re-render it if there is a configuration change
+    private val _currentMedia: MutableStateFlow<Media?> = MutableStateFlow(null)
+    val currentMedia: StateFlow<Media?> = _currentMedia
+
+    fun loadMedia() {
+        // Once we get a result from [TakePicture] and [TakeVideo], we clear the value of the
+        // temporaryMediaUri property
+        _temporaryMediaUri.value = null
+    }
+
+    // [TakePicture] and [TakeVideo] activityResult actions isn't returning the URI once it's
```

Comment:
nit: if you're using the link style for [TakePicture], etc... then make this a KDOC comment?